### ISSUE TITLE
Is this message by DeepSpeed indicating it is not using PyTorch-bundled CUDA?

### ISSUE THREAD

**Issue by XinliYu** [microsoft/DeepSpeed#3105]
By my understanding, DeepSpeed is complied and is using PyTorch-bundled CUDA.

Here is my PyTorch info when I execute the following,
> python -c 'import torch; print("torch version:", torch.__version__); print("cuda veersion:", torch.version.cuda); print("cuDNN version:", torch.backends.cudnn.version()); print("NCCL backend is available:", torch.distributed.is_nccl_available()); from scipy import optimize'

torch version: 2.0.0+cu117
cuda veersion: 11.7
cuDNN version: 8500
NCCL backend is available: True

However, when I run DeepSpeed, I see this message
>Installed CUDA version 11.1 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination

My system does have another CUDA version installed at /usr/local/cuda-11.1 , but I am not using it, and it is not in `PATH` and `LD_LIBRARY_PATH`. 

My Stanford Alpaca model fine-tuned with DeepSpeed is performing MUCH WORSE in comparison with the model fine-tuned without DeepSpeed. I am debugging, and I am concerned if the above message means DeepSpeed is not using the Pytorch bundled CUDA.

Thanks!

===========

Appendix - here is the full messages from DeepSpeed

>WARNING:root:Loading data...
WARNING:root:Formatting inputs...
WARNING:root:Loading data...
WARNING:root:Loading data...
WARNING:root:Tokenizing inputs... This may take some time...
WARNING:root:Loading data...
WARNING:root:Formatting inputs...
WARNING:root:Formatting inputs...
WARNING:root:Tokenizing inputs... This may take some time...
WARNING:root:Formatting inputs...
WARNING:root:Tokenizing inputs... This may take some time...
WARNING:root:Tokenizing inputs... This may take some time...
WARNING:root:Loading data...
WARNING:root:Loading data...
WARNING:root:Loading data...
WARNING:root:Formatting inputs...
WARNING:root:Formatting inputs...
WARNING:root:Formatting inputs...
WARNING:root:Tokenizing inputs... This may take some time...
WARNING:root:Tokenizing inputs... This may take some time...
WARNING:root:Tokenizing inputs... This may take some time...
WARNING:root:Loading data...
WARNING:root:Formatting inputs...
WARNING:root:Tokenizing inputs... This may take some time...
Installed CUDA version 11.1 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination
Installed CUDA version 11.1 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination
Installed CUDA version 11.1 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination
Installed CUDA version 11.1 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination
Installed CUDA version 11.1 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination
Installed CUDA version 11.1 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination
Installed CUDA version 11.1 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination
Installed CUDA version 11.1 does not match the version torch was compiled with 11.7 but since the APIs are compatible, accepting this combination
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Detected CUDA files, patching ldflags
Emitting ninja build file /storage/home/ec2-user/.cache/torch_extensions/py38_cu117/cpu_adam/build.ninja...
Building extension module cpu_adam...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
ninja: no work to do.
Loading extension module cpu_adam...
Time to load cpu_adam op: 1.049159049987793 seconds
Loading extension module cpu_adam...
Time to load cpu_adam op: 1.1045634746551514 seconds
Loading extension module cpu_adam...
Time to load cpu_adam op: 0.9584183692932129 seconds
Loading extension module cpu_adam...
Time to load cpu_adam op: 1.0704467296600342 seconds
Loading extension module cpu_adam...
Time to load cpu_adam op: 1.1197350025177002 seconds
Loading extension module cpu_adam...
Time to load cpu_adam op: 1.1285524368286133 seconds
Loading extension module cpu_adam...
Time to load cpu_adam op: 1.0468108654022217 seconds
Loading extension module cpu_adam...
Time to load cpu_adam op: 1.1659438610076904 seconds
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
Emitting ninja build file /storage/home/ec2-user/.cache/torch_extensions/py38_cu117/utils/build.ninja...
Building extension module utils...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
ninja: no work to do.
Loading extension module utils...
Time to load utils op: 0.4157865047454834 seconds
Loading extension module utils...
Time to load utils op: 0.4023163318634033 seconds
Loading extension module utils...
Time to load utils op: 0.4018673896789551 seconds
Loading extension module utils...
Time to load utils op: 0.4021468162536621 seconds
Loading extension module utils...
Time to load utils op: 0.5024268627166748 seconds
Loading extension module utils...
Loading extension module utils...
Time to load utils op: 0.5021953582763672 seconds
Time to load utils op: 0.5021004676818848 seconds
Loading extension module utils...
Time to load utils op: 0.5024056434631348 seconds
Rank: 0 partition count [16] and sizes[(421151488, False)] 
Rank: 7 partition count [16] and sizes[(421151488, False)] 
Rank: 3 partition count [16] and sizes[(421151488, False)] 
Rank: 4 partition count [16] and sizes[(421151488, False)] 
Rank: 5 partition count [16] and sizes[(421151488, False)] 
Rank: 6 partition count [16] and sizes[(421151488, False)] 
Rank: 2 partition count [16] and sizes[(421151488, False)] 
Rank: 1 partition count [16] and sizes[(421151488, False)] 
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
No modifications detected for re-loaded extension module utils, skipping build step...
Loading extension module utils...
Time to load utils op: 0.0006527900695800781 seconds
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
No modifications detected for re-loaded extension module utils, skipping build step...
Loading extension module utils...
Time to load utils op: 0.0004696846008300781 seconds
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
No modifications detected for re-loaded extension module utils, skipping build step...
Loading extension module utils...
Time to load utils op: 0.00044536590576171875 seconds
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
No modifications detected for re-loaded extension module utils, skipping build step...
Loading extension module utils...
Time to load utils op: 0.0004248619079589844 seconds
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
No modifications detected for re-loaded extension module utils, skipping build step...
Loading extension module utils...
Time to load utils op: 0.0004208087921142578 seconds
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
No modifications detected for re-loaded extension module utils, skipping build step...
Loading extension module utils...
Time to load utils op: 0.0004069805145263672 seconds
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
No modifications detected for re-loaded extension module utils, skipping build step...
Loading extension module utils...
Time to load utils op: 0.0004169940948486328 seconds
Using /storage/home/ec2-user/.cache/torch_extensions/py38_cu117 as PyTorch extensions root...
No modifications detected for re-loaded extension module utils, skipping build step...
Loading extension module utils...
Time to load utils op: 0.00038743019104003906 seconds
{'loss': 1.1965, 'learning_rate': 1.9998726619551668e-05, 'epoch': 0.25}                                                                                                                                                                                                                                                                                                                                                                                                                     
{'loss': 0.9874, 'learning_rate': 1.991568404156662e-05, 'epoch': 0.49}                                                                                                                                                                                                                                                                                                                                                                                                                      
{'loss': 0.9815, 'learning_rate': 1.9703936978669373e-05, 'epoch': 0.74}                                                                                                                                                                                                                                                                                                                                                                                                                     
{'loss': 0.9741, 'learning_rate': 1.9366233884375348e-05, 'epoch': 0.98}                                                                                                                                                                                                                                                                                                                                                                                                                     
{'loss': 0.7367, 'learning_rate': 1.89069581075914e-05, 'epoch': 1.23}                                                                                                                                                                                                                                                                                                                                                                                                                       
 21%|██████████████████████████████████████████████████████████████████████████████████████████▌

**Comment by HeyangQin** [comment]
Hi @XinliYu. Thank you for reporting the issue to us. Could you double-check the default `nvcc` by running `nvcc --version`? What is the output?
